### PR TITLE
refactor(feature-flags): share local evaluation helpers

### PR DESCRIPTION
**Pull request by marandaneto**
## Problem

The Node.js and Convex SDKs maintained separate implementations for scalar feature flag matching, deterministic bucketing, variant selection, and payload parsing. The copies could drift, but their full evaluators and SemVer behavior are not identical and should not be silently unified.

## Changes

- Add edge-safe local evaluation helpers to `@posthog/core`.
- Use the shared helpers from Node.js and Convex while keeping their evaluator orchestration local.
- Preserve the Node.js declaration build's DOM library reference after moving its SHA-1 implementation.
- Preserve strict Node.js SemVer matching and the existing permissive Convex policy explicitly.
- Add golden tests for matching, hashing, variant boundaries, payloads, and runtime-specific SemVer cases.

### Libraries affected

- [ ] All of them
- [ ] posthog-js (web)
- [ ] posthog-js-lite (web lite)
- [x] posthog-node
- [ ] posthog-react-native
- [ ] @posthog/react-native-plugin
- [ ] @posthog/react
- [ ] @posthog/ai
- [x] @posthog/convex
- [ ] @posthog/next
- [ ] @posthog/nextjs-config
- [ ] @posthog/nuxt
- [ ] @posthog/openfeature-node-provider
- [ ] @posthog/openfeature-web-provider
- [ ] @posthog/rollup-plugin
- [ ] @posthog/webpack-plugin
- [ ] @posthog/types
- [ ] @posthog/browser-common

## Checklist

- [x] Tests for new code
- [x] Accounted for the impact of any changes across different platforms
- [x] Accounted for backwards compatibility of any changes (no breaking changes!)
- [ ] Took care not to unnecessarily increase the bundle size

### If releasing new changes

- [x] Ran `pnpm changeset` to generate a changeset file

## 🤖 Agent context

**Autonomy:** Human-driven (agent-assisted)

Pi coding agent implemented and validated this refactor from a Slopo duplication report in a local session with no public transcript. The shared layer is intentionally limited to compatible primitives. Full evaluator parity and the Node.js/Convex SemVer differences remain out of scope.

### CODE REVIEW

**Review by dustinbyrne — APPROVED** [review]

File: packages/core/src/featureFlagLocalEvaluation.ts [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+
+export function getFeatureFlagVariantLookupTable(
+  variants: readonly FeatureFlagVariant[]
+): FeatureFlagVariantLookupEntry[] {
+  const table: FeatureFlagVariantLookupEntry[] = []
+  let valueMin = 0
+  for (const variant of variants) {
+    const valueMax = valueMin + variant.rollout_percentage / 100.0
+    table.push({ valueMin, valueMax, key: variant.key })
+    valueMin = valueMax
```
[the remark this answers is not in the document]
Comment (dustinbyrne) [reply packages/core/src/featureFlagLocalEvaluation.ts:350]: stop recommending this
